diff --git a/caffe-fast-rcnn/include/caffe/util/device_alternate.hpp b/caffe-fast-rcnn/include/caffe/util/device_alternate.hpp
index e3fe4fe..fd989e8 100644
--- a/caffe-fast-rcnn/include/caffe/util/device_alternate.hpp
+++ b/caffe-fast-rcnn/include/caffe/util/device_alternate.hpp
@@ -3,6 +3,7 @@
 
 #ifdef CPU_ONLY  // CPU-only Caffe.
 
+#include <algorithm>
 #include <vector>
 
 // Stub out GPU calls as unavailable.
@@ -86,7 +87,18 @@ const int CAFFE_CUDA_NUM_THREADS = 512;
 
 // CUDA: number of blocks for threads.
 inline int CAFFE_GET_BLOCKS(const int N) {
-  return (N + CAFFE_CUDA_NUM_THREADS - 1) / CAFFE_CUDA_NUM_THREADS;
+  //return (N + CAFFE_CUDA_NUM_THREADS - 1) / CAFFE_CUDA_NUM_THREADS;
+ static cudaDeviceProp *prop = NULL;
+ if (!prop) {
+    int device;
+    cudaGetDevice(&device);
+    prop = new cudaDeviceProp;
+    CUDA_CHECK(cudaGetDeviceProperties(prop, device));
+  } 
+
+  int num_blocks = (N + CAFFE_CUDA_NUM_THREADS - 1) / CAFFE_CUDA_NUM_THREADS;
+  int max_blocks = prop->maxGridSize[0];
+  return std::min<int>(num_blocks, max_blocks);
 }
 
 }  // namespace caffe
